### ISSUE TITLE
5.10 / 5.11 build fails

### ISSUE THREAD

**Issue by kronthto** [TuxInvader/focal-mainline-builder#4]
Hi,

thanks for the work you put into this, it works great (for 5.12.6) and is very transparent too.

However I cannot manage to build 5.10.39 (LTS) or 5.11.22 . 

Output from "your" script with `-e kver=v5.10.39`:
```
********

Renaming source package and updating control files

********
********

Applying default configs

********
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
debian/rules.d/0-common-vars.mk:15: *** first argument to 'word' function must be greater than 0.  Stop.
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
sed: can't read debian.master-21.10/changelog: No such file or directory
debian/rules.d/0-common-vars.mk:15: *** first argument to 'word' function must be greater than 0.  Stop.
********

Building packages
Command: dpkg-buildpackage --build=binary -aamd64 -d -uc -ui -us

********
dpkg-buildpackage: error: cannot open file debian/changelog: No such file or directory
********

Moving packages to debs folder

********
mv: cannot stat '/home/source/../*.*': No such file or directory
********

Cleaning git source tree
```

Actually the error is even similar to the log output of the official mainline builder that kept failing recently: 
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1928535/comments/3

So maybe this is not even an issue here but rather upstream in the ubuntu build scripts? 
I'm not too familiar with the whole kernel build process so I hope it's OK I opened this issue here to ask.

**Comment by kronthto** [comment]
When simply changing:

```diff
--- a/debian/debian.env
+++ b/debian/debian.env
@@ -1 +1 @@
-DEBIAN=debian.master-21.10
+DEBIAN=debian.master
```

it passes. In 5.12 it is that way from the beginning, and also has been until 5.10.37.


Apparently this file is generated by the ubuntu kernel bot new every release:

5.10.37:
```diff
commit f3cf7622efde3cd18d3f4e70ee2bd69f8c764492
Author: Kernel Builder (kathleen) <ubuntu-kernel-bot@canonical.com>
Date:   Fri May 14 09:34:35 2021 +0000

    base packaging

diff --git a/debian/debian.env b/debian/debian.env
new file mode 100644
index 000000000000..841bc27141e4
--- /dev/null
+++ b/debian/debian.env
@@ -0,0 +1 @@
+DEBIAN=debian.master-21.10
```
5.12.6:
```diff
commit 8e019e1f614023866e34a9b6b0b54ac6ab7cbf20
Author: Kernel Builder (sita) <ubuntu-kernel-bot@canonical.com>
Date:   Sat May 22 10:33:33 2021 +0000

    base packaging

diff --git a/debian/debian.env b/debian/debian.env
new file mode 100644
index 000000000000..be31a0c27019
--- /dev/null
+++ b/debian/debian.env
@@ -0,0 +1 @@
+DEBIAN=debian.master
```
And up to 5.10.36 it was also still "good":
```diff
commit f0cc96561db7d294a8d39833e1ba90677191d18c
Author: Kernel Builder (sita) <ubuntu-kernel-bot@canonical.com>
Date:   Tue May 11 13:31:18 2021 +0000

    base packaging

diff --git a/debian/debian.env b/debian/debian.env
new file mode 100644
index 000000000000..be31a0c27019
--- /dev/null
+++ b/debian/debian.env
@@ -0,0 +1 @@
+DEBIAN=debian.master
```

I also reported this in https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1928535/comments/9

**Comment by TuxInvader** [comment]
Hi Kronthto

You're absolutely right. That does seem to be the cause of the build failures on mainline, thanks for investigating. I've added an undocumented `--checkbugs=yes` flag to the build script. If you pass this flag it will check for any known bugs, which currently is the `debian.env` issue you have identified and attempt to workaround it.

See: c676b68d2d6ff80bac67ccda24e78a0e90d938b8

I've tested with 3.10.39, and it seems to build fine.

Thanks!!!
